ui/shared/RevisionList.jsx

```diff
@@ -4,9 +4,41 @@ import { Col, Row } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faExternalLinkSquareAlt } from '@fortawesome/free-solid-svg-icons';
 
+import { getData } from '../helpers/http';
+import { bugzillaBugsApi } from '../helpers/url';
+
 import { Revision } from './Revision';
 
 export class RevisionList extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      revisionComments: {},
+    };
+  }
+
+  componentDidMount() {
+    const { revisions } = this.props;
+    revisions.forEach((revision, i) => {
+      this.getRevisionComment(revision.comments, i);
+    });
+  }
+
+  getRevisionComment = async (comments) => {
+    const comment = comments.split('\n')[0];
+    const bugMatches = comment.match(/-- ([0-9]+)|bug.([0-9]+)/gi);
+    const bugNumbers = bugMatches.map((bugMatch) => bugMatch.split(' ')[1]);
+    const { data } = await getData(bugzillaBugsApi('bug', { id: bugNumbers }));
```

Comment:
This code ends up calling the `bugzillaBugsApi` too often.  We should batch up the bug numbers and only call that API once for all of them.

I'm trying to consider what the most efficient way to do this with multiple pushes.  Each `Push` has a `RevisionList` which has one or more `Revision`s.  So the most efficient approach would be to gather all the bug numbers for all `Revision`s of all `RevisionList`s of all `Push`s.

It may be good enough to make a bug API call on each `RevisionList`, as long as we keep that in the `revisionComments` and only make that call once.  If that ends up being too sluggish when you fetch `50` pushes, we may need to move this up into the `Redux` code.  It would make it slightly more complicated, but I can help you with that.  :)

Reply:
Instead of calling `bugzillaBugsApi` every `Revision`, I'm now gathering all bug numbers in a `RevisionList` and calling `bugzillaBugsApi` at the end.

I saw what you meant by too sluggish. It took ~15-20 seconds when hitting "get next" at 50 with this change.

Reply:
Instead of calling bugzillaBugsApi every Revision, I'm now gathering all bug numbers in a RevisionList and calling bugzillaBugsApi at the end.

I see what you mean by too sluggish. It took a while to load when hitting "get next" at 50 with this change.